File: app/src/main/java/com/example/foodieempire/view/register/LoginFragment.java
package com.example.foodieempire.view.register;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.example.foodieempire.R;
import com.example.foodieempire.databinding.FragmentLoginBinding;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

public class LoginFragment extends Fragment {
    FragmentLoginBinding binding;
    FirebaseAuth firebaseAuth = FirebaseAuth.getInstance();

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View loginFragment = inflater.inflate(R.layout.fragment_login, container, false);
        binding = FragmentLoginBinding.bind(loginFragment);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding.loginBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String mail = binding.mailET.getText().toString();
                String pass = binding.passET.getText().toString();
                if (isDataEmpty(mail, pass)) {
                    Toast.makeText(getActivity(), "Please Complete Your Data", Toast.LENGTH_SHORT).show();
                } else {
                    loginWithFirebaseAuth(mail, pass);
                }
            }
        });
    }

    private void loginWithFirebaseAuth(String mail, String pass) {
        firebaseAuth.signInWithEmailAndPassword(mail, pass).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
            @Override
            public void onComplete(@NonNull Task<AuthResult> task) {
                if (task.isSuccessful()) {
                    Navigation.findNavController(binding.getRoot()).navigate(R.id.action_loginFragment_to_mainActivity);
                    getActivity().finish();
                } else {
                    Toast.makeText(getActivity(), task.getException().getMessage(), Toast.LENGTH_SHORT).show();
                }
            }
        });
    }

    private boolean isDataEmpty(String mail, String pass) {
        return mail.isEmpty() || pass.isEmpty();
    }
}

File: app/src/main/java/com/example/foodieempire/view/register/LoginFragment.kt
package com.example.foodieempire.view.register

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.findNavController
import com.example.foodieempire.R
import com.example.foodieempire.controller.AppSharedPreference
import com.example.foodieempire.databinding.FragmentLoginBinding
import com.google.firebase.auth.FirebaseAuth

class LoginFragment : Fragment() {
    var binding: FragmentLoginBinding? = null
    var firebaseAuth = FirebaseAuth.getInstance()
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val loginFragment = inflater.inflate(R.layout.fragment_login, container, false)
        binding = FragmentLoginBinding.bind(loginFragment)
        return binding!!.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding!!.loginBtn.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View) {
                val mail = binding!!.mailET.text.toString()
                val pass = binding!!.passET.text.toString()
                val name = binding!!.nameET.text.toString()
                if (isDataEmpty(mail, pass, name)) {
                    Toast.makeText(activity, "Please Complete Your Data", Toast.LENGTH_SHORT).show()
                } else {
                    loginWithFirebaseAuth(mail, pass)
                    writeToSharedPref(name)
                }
            }

            private fun writeToSharedPref(name: String) {
                AppSharedPreference.writeToSharedPrefernce(activity, name)
            }
        })
    }

    private fun loginWithFirebaseAuth(mail: String, pass: String) {
        firebaseAuth.signInWithEmailAndPassword(mail, pass).addOnCompleteListener { task ->
            if (task.isSuccessful) {
                binding!!.loginBtn.visibility = View.GONE
                binding!!.loginProgress.visibility = View.VISIBLE
                findNavController(binding!!.root).navigate(R.id.action_loginFragment_to_mainActivity)
                requireActivity().finish()
            } else {
                binding!!.loginProgress.visibility = View.GONE
                Toast.makeText(activity, task.exception!!.message, Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun isDataEmpty(mail: String, pass: String, name: String): Boolean {
        return mail.isEmpty() || pass.isEmpty() || name.isEmpty()
    }
}